### DISCUSSION TITLE
Perpendicular link out of port

### DISCUSSION THREAD

**Discussion by Timot05** [clientIO/joint#2251] [Q&A] [answered]
### Introduction

Hi, 
I'm trying to generate a link between a port on one element and a virtual point that is offset from the port and perpendicular to the edge of that element. I've been trying to utilize the custom anchor without success (https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#anchors.custom). Would you have advice on the best way to achieve this effect?

Thanks!

### Restrictions & Constraints

The link should move alongside the element when the element is moved.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
Do you mean something like [this](https://jsfiddle.net/kumilingus/mos6a5w1/)? If not please elaborate.

**Reply by Timot05** [reply]
This is exactly what I meant. Thank you. Is there a way to extract the direction automatically based on where the port is located? That way the link would be pointed towards the right if the port is on the right, towards the bottom if the port is on the bottom, etc...

**Reply by kumilingus** [reply]
That's a job for a custom anchor: https://jsfiddle.net/kumilingus/8h4svazx/

**Reply by Timot05** [reply]
Thanks @kumilingus! That seems to work in case there is a single port spread out around the cell but not if there are individual ports for each side of the cell. Do you have a workaround if that is the case?

**Reply by kumilingus** [reply]
I don't think I understand. Please update the example to show the issue.

Here's an element with multiple ports per side: https://jsfiddle.net/kumilingus/8h4svazx/18/

**Reply by Timot05** [reply]
Just saved a new version. Both links are overlayed on top of each other in this case. But the intent "bottom" facing down.

**Reply by kumilingus** [reply]
Where can I find the new version?

**Reply by Timot05** [reply]
Sorry, thought I saved it. This should work: https://jsfiddle.net/qnjc5dk6/

**Reply by kumilingus** [reply]
Why did you change the markup of the ports? The new markup does not have the `portBody` selector. 

https://jsfiddle.net/kumilingus/903grq4a/

**Reply by Timot05** [reply]
That was the issue indeed. Thanks for the support. Very appreciated!
